src/org/zaproxy/zap/extension/params/ExtensionParams.java

```diff
@@ -305,6 +307,16 @@ public boolean onHttpRequestSend(HttpMessage msg) {
 			persist(sps.addParam(site, param, msg));
 		}
 		
+		VariantMultipartFormParameters params2 = new VariantMultipartFormParameters();
+		params2.setMessage(msg);
+		for (NameValuePair nvp : params2.getParamList()) {
+			if (Arrays.asList(NameValuePair.TYPE_MULTIPART_DATA_PARAM, NameValuePair.TYPE_MULTIPART_DATA_FILE_NAME)

```

Comment:
This could be created once before the for loop (or moved to a constant, or just compare the types).

Reply:
Changed it to compare `nvp.getType()` vs the two types (ORed).

Reply:
Looks good.